GitUI/CommandsDialogs/SettingsDialog/Pages/ScriptsSettingsPage.cs

```diff
@@ -151,20 +166,20 @@ public override void OnPageShown()
             // dummy request; for some strange reason the ResourceSets are not loaded until after the first object request... bug?
             rm.GetObject("dummy");
 
-            using System.Resources.ResourceSet resourceSet = rm.GetResourceSet(CultureInfo.CurrentUICulture, true, true);
+            using System.Resources.ResourceSet resourceSet = rm.GetResourceSet(CultureInfo.CurrentUICulture, true, true)!;
             foreach (DictionaryEntry icon in resourceSet.Cast<DictionaryEntry>().OrderBy(icon => icon.Key))
             {
                 if (icon.Value is Bitmap bitmap)
                 {
-                    EmbeddedIcons.Images.Add(icon.Key.ToString(), bitmap.AdaptLightness());
+                    EmbeddedIcons.Images.Add(icon.Key.ToString()!, bitmap.AdaptLightness());
```

Comment:
ditto

Reply:
I've watched some MS dev discuss how they've gone about adding nullable annotations on the core lib and `ToString` was one of those where they **had** to mark it as nullable because someone might override it and return null i.e. they don't control it. But in this case where we know the keys to be not null as  ourselves put non-nullable items into this dictionary can mark it with `!`. I am not sure if `Validates.NotNull` is necessary here or in any other place where we are sure of what's in the thing we are calling but don't actually control nullability annotations. Thoughts?

Reply:
OK, this time, SourceLink has pointed to:
```
        public virtual string? ToString()
```
One hour ago, I saw a non-null `ToString` - most likely not of `object`.

Yes, only the .NET runtime can protect from `return null!`.
`Validates.NotNull` makes it explicit that a dev's assumption was wrong or the code was broken by a change.
A bug report with an NRE does not become clear until one analyzes the callstack.

Reply:
Do you have any scenario in mind where using a `!` is warranted? I wonder if there is a _potential-null-ref-analyzer_ for VS because in Rider basically any GE file is littered with these warnings and someone using only VS without additional analyzers has no clue how many places can get nulls according to types while in reality that would never happen due to how code is used.

There are cases where potential-null is passed, where `?` is used when it's know not to be null, etc. I tend to fix these, not blindly, but by evaluating when a thing can actually be null if we mess up or when it's simply not possible for the compiler to see that null is not possible thus a `!` is fine.
Look:
![image](https://github.com/gitextensions/gitextensions/assets/483659/fe435cfc-4c58-419d-9e6b-df4ecc361db6)

Reply:
We started to enforce "?", i.e. "#nullable enable", for some subprojects and for single files - unfinished.
"!" is quickly written, and it is easy to miss that it is there.
With the recent change, we are pretty safe that "!" is OK here.